src/dump.c

```diff
@@ -1298,7 +1298,10 @@ static void jl_collect_backedges( /* edges */ jl_array_t *s, /* ext_targets */ j
     for (i = 0; i < edges_map.size; i += 2) {
         jl_method_instance_t *caller = (jl_method_instance_t*)table[i];
         jl_array_t *callees = (jl_array_t*)table[i + 1];
-        if (callees != HT_NOTFOUND && (module_in_worklist(caller->def.method->module) || method_instance_in_queue(caller))) {
+        if (callees == HT_NOTFOUND)
+            continue;
+        assert(jl_is_method_instance(caller));
+        if (jl_is_method(caller->def.method) && (module_in_worklist(caller->def.method->module) || method_instance_in_queue(caller))) {
```

Comment:
AFIAK, we should never see this happen (since `Core.Compiler.store_backedges` shouldn't make these edges). Though we could also have `jl_method_instance_add_backedge` ensure that it does not create them too.
```suggestion
        assert(jl_is_method_instance(caller) && jl_is_method(caller->def.method));
        if (module_in_worklist(caller->def.method->module) || method_instance_in_queue(caller))) {
```